src/main/resources/application.conf

```diff
@@ -198,26 +198,49 @@ mantis {
     transaction-timeout = 2.minutes
   }
 
-  mining {
-    # Maximum number of ommers kept in the pool
-    ommers-pool-size = 30
+  consensus {
+    # FIXME this was moved here from the `mining` section
+    active-timeout = 5 seconds
 
-    # This determines how many parallel eth_getWork request we can handle, by storing the prepared blocks in a cache,
-    # until a corresponding eth_submitWork request is received
-    block-cashe-size = 30
+    # New timeout to replace `ommer-pool-query-timeout` for uses outside ethash mining
+    get-transaction-from-pool-timeout = 5.seconds
 
+    # Moved from the `mining` section
     # Miner's coinbase address
     coinbase = "0011223344556677889900112233445566778899"
 
-    active-timeout = 5 seconds
-
-    ommer-pool-query-timeout = 5.seconds
-
+    # Moved from the `mining` section
     # Extra data to add to mined blocks
     header-extra-data = "mantis"
 
-    # If true this node will mine new blocks
+    #####################
+    # Moved from the `mining` section because this is needed by BlockGenerator, which we use in other consensus
+    # protocols as well.
+    #####################
+    # This determines how many parallel eth_getWork request we can handle, by storing the prepared blocks in a cache,
+    # until a corresponding eth_submitWork request is received
+    block-cashe-size = 30
+
+    # See io.iohk.ethereum.consensus.Protocol for the available protocols.
+    # Declaring the protocol here means that a more protocol-specific configuration
+    # is pulled from the corresponding consensus implementation.
+    # For example, in case of ethash, a section named `mining` is used.
+    protocol = ethash
+
+    # If true then the consensus protocol uses this node for mining.
+    # In the case of ethash PoW, this means mining new blocks, as specified by Ethereum.
+    # In the general case, the semantics are due to the specific consensus implementation.
     mining-enabled = false
+  }
```

Comment:
Even if they are not used, protocol specific settings (Raft) should be added here, because this file serves also as a developer's reference - so we know what settings are possible. This can be added in the next iteration though.